pandasai/response/response_parser.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from typing import Any
+from pandasai.helpers.env import is_running_in_console
+from ..helpers.df_info import polars_imported
+
+from pandasai.exceptions import MethodNotImplementedError
+
+
+class IResponseParser(ABC):
+    @abstractmethod
+    def parse(self, result: dict) -> Any:
+        """Runs the parser"""
+        raise MethodNotImplementedError
+
+
+class ResponseParser(IResponseParser):
+    _context = None
+
+    def __init__(self, context) -> None:
+        self._context = context
+
+    def parse(self, result) -> Any:
+        if result["type"] == "dataframe":
+            return self.format_dataframe(result)
+        elif result["type"] == "plot":
+            return self.format_plot(result)
+        else:
+            return self.format_other(result)
+
+    def format_dataframe(self, result: dict) -> Any:
+        # Format and process dataframe
+        from ..smart_dataframe import SmartDataframe
```

Comment:
Let's maybe validate that `result[value]` is set. What do you think?

Reply:
@gventuri   I think its something extra to validate here again because we are making sure in our chat function when we pass result to parser.parse function it has the result.